src/routes/coupon.php

```diff
@@ -35,18 +36,19 @@
     } catch (PDOException $e) {
         $db = null;
         $errorMessage = $e->getMessage();
-        return $response->write(json_encode(['error'=>$errorMessage]))->withStatus(500);
+        return $response->write(json_encode(['error' => $errorMessage]))->withStatus(500);
     }
 });
 
 //Get Usable Coupons By User and Hotel Sort By Expire Date
 $app->get('/api/coupons/customer/{username}/hotel/{id}', function (Request $request, Response $response, array $args) {
     $username = $args['username'];
     $hotel_id = $args['id'];
-    $current=date('Y-m-d');
-    $sql = "SELECT C.id, C.username, C.type_id, C.hotel_id, C.expire_date, CT.value, CT.discount_type 
+    $current = date('Y-m-d');
+    $sql = "SELECT C.id, C.username, C.coupon_type_id, C.hotel_id, C.expire_date, CT.value, CT.discount_type 
             FROM coupon C, coupon_type CT 
-            WHERE username = :username AND hotel_id = :hotel_id AND C.type_id = CT.id AND C.expire_date >= :current
+            WHERE username = :username AND hotel_id = :hotel_id AND C.coupon_type_id = CT.id AND C.expire_date >= :current
+            AND c.is_used = 0 
             AND C.id NOT IN ( SELECT coupon_id FROM payment WHERE coupon_id IS NOT NULL)
```

Comment:
If we have the `is_used` to filter all the used coupon, do we still need `AND C.id NOT IN ( SELECT coupon_id FROM payment WHERE coupon_id IS NOT NULL)` ?

Reply:
After reading the code, I do not think `is_used` is a useful field.
This field is not used in the front-end because the front-end only wants the not-used coupons (we only show not-used coupons, and I do not think we need to show the used ones).
And the old query can already find all the coupons that are not-used.